### ISSUE TITLE
nodejs: illegal instruction

### ISSUE THREAD

**Issue by sobercoder** [archlinuxarm/PKGBUILDs#1103]
Package: nodejs-0.12.0-1
Architecture: armv6h
Device: raspberry pi

_What it isn't doing:_
When the node binary is executed, it gives the message __Illegal instruction (core dumped)__.
When run with the `--version` switch, it does show the correct version.

_What it should be doing:_
Executing as usual providing a prompt.

_Steps to reproduce:_
Run the `node` command with no arguments.

**Comment by silverwind** [comment]
This was fixed in io.js 1.0.4, details [here](https://github.com/iojs/io.js/issues/283). I'd appreciate getting node.js replaced by io.js. They provide ARMv6 binaries too.

**Comment by WarheadsSE** [comment]
We're aware that it has been addressed in io.js, and that it will be fixed in the upcoming version of node.js. We will not, however be replacing node.js. 

Also, io.js providing binaries does nothing for The Arch Way, and has no impact to the distribution.

**Comment by WarheadsSE** [comment]
I have made comment over at the actual node.js Issues (https://github.com/joyent/node/pull/9068)

**Comment by bendo** [comment]
0.12.1 was released ;)

**Comment by yar0d** [comment]
A core dump occurs with this release too.

Package: community/nodejs 0.12.1-1
Architecture: ARMv7l PJ4
Device: Cubox - http://archlinuxarm.org/platforms/armv7/marvell/cubox

$ node --version
v0.12.1
$ node
Illegal instruction (core dumped)

**Comment by silverwind** [comment]
@yar0d the fix hasn't landed yet in node, it targets 0.12.2 currently: https://github.com/joyent/node/pull/14193

**Comment by WarheadsSE** [comment]
They have release 0.12.2, and still not fixed this. We will continue to push.

**Comment by yar0d** [comment]
I hope that 0.12.3 will solve this :-)

**Comment by silverwind** [comment]
We recently uncovered another armv6 SIGILL bug in v8: https://github.com/iojs/io.js/issues/1376

**Comment by qjcg** [comment]
Package: nodejs-0.12.7-1
Architecture: armv7
Device: Odroid-XU3-lite, Odroid-C1

Running node to drop into the interpreter runs fine for me with the package version / hardware above.

```javascript
$ node
> console.log("Time to close issue #1103?");
Time to close issue #1103?
undefined
>
```
Is anyone still experiencing this issue?

**Comment by silverwind** [comment]
@qjcg you're on ARMv7, this issue is about ARMv6.

node v4.0.0 will be available in a few days which will finally resolve this.

**Comment by qjcg** [comment]
@silverwind, fair enough, I wasn't sure since @yar0d mentioned running into this on armv7 in his post above.

**Comment by silverwind** [comment]
@qjcg Oh right, I missed that. For all I know, this never affected ARMv7 devices like RPi2 or Odroids. Maybe ARMv7l PJ4 is giving ARMv6-like `cpuinfo` and was hit by this bug as well, I'd suggest some testing with a recent io.js/node.js version on that device.

**Comment by silverwind** [comment]
node v4.0.0 is now available which definitely fixes this.

**Comment by kmihelich** [comment]
Fixed in 4.0.0-1.
